src/ResourceManager/Sql/Commands.Sql/Elastic Jobs/Services/AzureSqlElasticJobAdapter.cs

```diff
@@ -0,0 +1,1164 @@
+        }
+
+        /// <summary>
+        /// Gets a root job execution
+        /// </summary>
+        /// <param name="resourceGroupName">The resource group</param>
+        /// <param name="serverName">The server the agent is in</param>
+        /// <param name="agentName">The agent name</param>
+        /// <returns>The converted agent model</returns>
+        public AzureSqlElasticJobExecutionModel GetJobExecution(string resourceGroupName, string serverName, string agentName, string jobName, Guid jobExecutionId)
+        {
+            var resp = Communicator.GetJobExecution(resourceGroupName, serverName, agentName, jobName, jobExecutionId);
+            return CreateJobExecutionModelFromResponse(resourceGroupName, serverName, agentName, jobName, resp);
+        }
+
+        /// <summary>
+        /// Cancels a job execution
+        /// </summary>
+        /// <param name="resourceGroupName">The resource group name</param>
+        /// <param name="serverName">The server the agents are in</param>
+        /// <returns>The converted agent model(s)</returns>
+        public void CancelJobExecution(AzureSqlElasticJobExecutionModel model)
+        {
+            Communicator.CancelJobExecution(model.ResourceGroupName, model.ServerName, model.AgentName, model.JobName, model.JobExecutionId.Value);
+        }
+
+        /// <summary>
+        /// Deletes an Azure SQL Database Agent associated to a server
+        /// </summary>
+        /// <param name="resourceGroupName">The resource group name</param>
+        /// <param name="serverName">The server the agent is in</param>
+        /// <param name="agentName">The agent name</param>
+        public List<AzureSqlElasticJobExecutionModel> ListByAgent(
+            string resourceGroupName,
+            string serverName,
+            string agentName,
+            DateTime? createTimeMin = null,
+            DateTime? createTimeMax = null,
+            DateTime? endTimeMin = null,
+            DateTime? endTimeMax = null,
+            bool? isActive = null,
+            int? skip = null,
+            int? top = null)
+        {
+            IPage<JobExecution> resp = Communicator.ListJobExecutionsByAgent(
+                resourceGroupName,
+                serverName,
+                agentName,
+                createTimeMin,
+                createTimeMax,
+                endTimeMin,
+                endTimeMax,
+                isActive,
+                skip,
+                top);
+
+            List<AzureSqlElasticJobExecutionModel> executions = resp.Select((jobExecution) =>
+                CreateJobExecutionModelFromResponse(resourceGroupName, serverName, agentName,
+                new ResourceIdentifier(jobExecution.Id).ParentResourceBuilder[5], jobExecution)).ToList();
+
+            if (resp.NextPageLink != null)
+            {
+                Match match = Regex.Match(resp.NextPageLink, @"skip=(\d*)");
+                int toSkip = int.Parse(match.Groups[1].Value);
```

Comment:
If our API works correctly then you shouldn't need to Parse the next page link like this. You should be able to just keep getting until NextPageLink is null. Server should do the math to determine based on top/skip whether or not to give you a NextPageLink.

Reply:
Gotcha. Will update this to perform GETs until next page link is null instead without all this fancy math and regex parsing for skip.